test/performance/specs/site-editor.spec.js

```diff
@@ -0,0 +1,206 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+/**
+ * External dependencies
+ */
+const path = require( 'path' );
+
+/**
+ * Internal dependencies
+ */
+const {
+	readFile,
+	deleteFile,
+	saveResultsFile,
+	getTraceFilePath,
+	getTypingEventDurations,
+	getLoadingDurations,
+	loadBlocksFromHtml,
+} = require( '../utils' );
+
+const results = {
+	serverResponse: [],
+	firstPaint: [],
+	domContentLoaded: [],
+	loaded: [],
+	firstContentfulPaint: [],
+	firstBlock: [],
+	type: [],
+	typeContainer: [],
+	focus: [],
+	inserterOpen: [],
+	inserterHover: [],
+	inserterSearch: [],
+	listViewOpen: [],
+};
+
+let testPageId;
+
+test.describe( 'Site Editor Performance', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'emptytheme' );
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		saveResultsFile( __filename, results );
+
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	test( 'Loading', async ( { browser, page, admin } ) => {
+		// Start a new page.
+		await admin.createNewPost( { postType: 'page' } );
+
+		// Turn the large post HTML into blocks and insert.
+		await loadBlocksFromHtml(
+			page,
+			path.join( process.env.ASSETS_PATH, 'large-post.html' )
+		);
+
+		// Save the draft.
+		await page
+			.getByRole( 'button', { name: 'Save draft' } )
+			.click( { timeout: 60_000 } );
+		await expect(
+			page.getByRole( 'button', { name: 'Saved' } )
+		).toBeDisabled();
+
+		// Get the ID of the saved page.
+		testPageId = await page.evaluate( () =>
+			new URL( document.location ).searchParams.get( 'post' )
+		);
+
+		// Open the test page in Site Editor.
+		await admin.visitSiteEditor( {
+			postId: testPageId,
+			postType: 'page',
+		} );
+
+		// Get the URL that we will be testing against.
+		const targetUrl = page.url();
+
+		// Start the measurements.
+		let i = 3;
+		while ( i-- ) {
+			// Open a fresh page in a new context to prevent caching.
+			const testPage = await browser.newPage();
```

Comment:
It's recommended to explicitly create a new context and a page from that context. See https://playwright.dev/docs/api/class-browser#browser-new-page.

The creation of the new context might lazily evaluate something heavy for performance too, which might explain the performance drop in some senerios? We can try spinning up a new page and closing it immediately to warm it up a little bit.

Reply:
I like the image of warming up the page by reloads. 😄 I'll try to initialize the context separately - maybe we'll get closer to the original numbers! 🤞